scripts/update-and-check-binary.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+BASE_URL="https://binaries.soliditylang.org/bin"
+REPO_ROOT="$(dirname "$0")/.."
+
+function fail() {
+	echo -e "$@" >&2
+	exit 1
+}
+
+function check_version() {
+	current_version=$(node ./dist/solc.js --version | sed -En 's/^(.*).Emscripten.*/\1/p')
+
+	# Retrieve the correspondent released version
+	short_version=$(echo "${current_version}" | sed -En 's/^([0-9.]+).*\+commit\.[0-9a-f]+.*$/\1/p')
+	release_version=$(curl -s "${BASE_URL}/list.json"  | jq ".releases | .[\"${short_version}\"]" | tr -d '"' | sed -En 's/^soljson-v(.*).js$/\1/p')
+
+	# check if current version exists as release
+	if [ "${current_version}" != "${release_version}" ]; then
+		fail "Failed: version mismatch:\n [current]: ${current_version}\n [release]: ${release_version}"
+	fi
+
+	current_sha=$(shasum -b -a 256 ./soljson.js | awk '{ print $1 }')
+	release_sha=$(curl -s "${BASE_URL}/list.json" | jq ".builds[] | select(.longVersion == \"${release_version}\") | .sha256" | tr -d '"' | sed 's/^0x//')
+
+	# check if sha matches
+	if [ "${current_sha}" != "${release_sha}" ]; then
+		fail "Failed: checksum mismatch:\n [current]: ${current_sha}\n [release]: ${release_sha}"
+	fi
+
+	# check if the current version is the latest release
+	latest_version=$(curl -s "${BASE_URL}/list.json"  | jq ".latestRelease" | tr -d '"')
+	if [ "${short_version}" != "${latest_version}" ]; then
```

Comment:
I think it would be best to compare the full version, including the commit hash.

Reply:
Yes, I do that in the first check. The 3 checks are:
1. check if the currently installed version is a known release (it may not be the latest)
2. check if the sha of both matches
3. check if the current is also the latest

Indeed the first check seems useless, and I will just check the 2 and 3, using the commit hash in 3.